update.go

```diff
@@ -218,46 +220,81 @@ func (u *Update) downloadFile() error {
 	return nil
 }
 
-// getBinPath get bin folder path.
-func (u *Update) getBinPath(envPath, homeDir string) string {
-	binPath := ""
-
+// setBinPath get bin folder path.
+func (u *Update) setBinPath(envPath, homeDir string) {
 	if strings.Contains(envPath, homeDir+"/.global/bin") {
-		binPath = filepath.Join(homeDir, ".global/bin")
+		u.fDir = filepath.Join(homeDir, ".global/bin")
 	} else if strings.Contains(envPath, homeDir+"/.local/bin") {
-		binPath = filepath.Join(homeDir, ".local/bin")
+		u.fDir = filepath.Join(homeDir, ".local/bin")
 	} else if strings.Contains(envPath, "/usr/local/bin") {
-		binPath = "/usr/local/bin"
+		u.fDir = "/usr/local/bin"
 	}
 
-	u.fPath = strings.TrimSpace(filepath.Join(binPath, u.fName))
-	return binPath
+	u.fPath = strings.TrimSpace(filepath.Join(u.fDir, u.fName))
 }
 
 // installFile copy file to the bin folder and remove temp file.
 func (u *Update) installFile(dirPath string) error {
 	cli.Println("Installing %s binary under %s", u.fName, dirPath)
 
-	srcFile, err := os.Open(u.fTmpPath)
+	info, err := os.Stat(u.fDir)
+	if err != nil {
+		return err
+	}
+	pathPerm := fmt.Sprintf("%04o", info.Mode().Perm())
+
+	// Copy temp file in the plasmactl folder.
+	src, err := os.Open(u.fTmpPath)
 	if err != nil {
 		return err
 	}
-	defer srcFile.Close()
 
-	destFile, err := os.Create(u.fPath)
+	// Set temp permissions for the folder with plasmactl.
+	if err = u.setFolderPermissions("777", u.fDir); err != nil {
+		return err
+	}
+
+	fTmpName := u.fPath + ".tmp"
+	dst, err := os.Create(fTmpName)
 	if err != nil {
+		src.Close()
+		u.setFolderPermissions(pathPerm, u.fDir)
 		return err
 	}
-	defer destFile.Close()
 
-	if _, err = io.Copy(destFile, srcFile); err != nil {
+	_, err = io.Copy(dst, src)
+	src.Close()
+	dst.Close()
+	if err != nil {
+		u.setFolderPermissions(pathPerm, u.fDir)
 		return err
 	}
 
-	if err = os.Remove(u.fTmpPath); err != nil {
+	// Rename temp file to plasmactl.
+	if err = os.Rename(fTmpName, u.fPath); err != nil {
+		log.Debug("Failed to rename temp file.")
+		u.setFolderPermissions(pathPerm, u.fDir)
 		return err
 	}
 
+	// Set plasmactl permissions.
+	if err = u.setFolderPermissions("755", u.fPath); err != nil {
+		return err
+	}
+	// Get back folder permissions.
+	if err = u.setFolderPermissions(pathPerm, u.fDir); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (u *Update) setFolderPermissions(pathPerm, fPath string) error {
+	cmd := exec.Command("sudo", "chmod", pathPerm, fPath)
```

Comment:
Better use a library to change permissions instead of exec.Command()

Also this creates dependency to `sudo`, which may not be present on users local

Reply:
Looks like no library to do that actually, i'll create a follow up issue to make it better though